File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/mangaextractor/MangaVolume.java
package org.jisho.textosJapones.model.entities.mangaextractor;

import com.google.gson.annotations.Expose;
import org.jisho.textosJapones.model.entities.Manga;
import org.jisho.textosJapones.model.entities.VocabularioExterno;
import org.jisho.textosJapones.model.enums.Language;

import java.util.*;

public class MangaVolume extends Manga {

	private UUID id;
	@Expose
	private String manga;
	@Expose
	private Integer volume;
	@Expose
	private Language lingua;
	@Expose
	private List<MangaCapitulo> capitulos;
	@Expose
	private Set<VocabularioExterno> vocabularios;
	@Expose
	private String arquivo;
	private Boolean processado;

	private MangaCapa capa;

	public UUID getId() {
		return id;
	}

	public void setId(UUID id) {
		this.id = id;
	}

	public String getManga() {
		return manga;
	}

	public void setManga(String manga) {
		this.manga = manga;
	}

	public Integer getVolume() {
		return volume;
	}

	public void setVolume(Integer volume) {
		this.volume = volume;
	}

	public Language getLingua() {
		return lingua;
	}

	public void setLingua(Language lingua) {
		this.lingua = lingua;
	}

	public String getArquivo() {
		return arquivo;
	}

	public void setArquivo(String arquivo) {
		this.arquivo = arquivo;
	}

	public List<MangaCapitulo> getCapitulos() {
		return capitulos;
	}

	public void setCapitulos(List<MangaCapitulo> capitulos) {
		this.capitulos = capitulos;
	}

	public void addCapitulos(MangaCapitulo capitulo) {
		this.capitulos.add(capitulo);
	}

	public Set<VocabularioExterno> getVocabularios() {
		return vocabularios;
	}

	public void setVocabularios(Set<VocabularioExterno> vocabularios) {
		this.vocabularios = vocabularios;
	}

	public void addVocabulario(VocabularioExterno vocabulario) {
		this.vocabularios.add(vocabulario);
	}

	public Boolean getProcessado() {
		return processado;
	}

	public void setProcessado(Boolean processado) {
		this.processado = processado;
	}

	public MangaCapa getCapa() {
		return capa;
	}

	public void setCapa(MangaCapa capa) {
		this.capa = capa;
	}

	public MangaVolume() {
		super();
		this.id = null;
		this.manga = "";
		this.volume = 0;
		this.lingua = Language.PORTUGUESE;
		this.vocabularios = new HashSet<VocabularioExterno>();
		this.processado = false;
		this.processar = true;
		this.capitulos = new ArrayList<MangaCapitulo>();
		this.arquivo = "";
		this.capa = null;
	}

	public MangaVolume(UUID id, String manga, Integer volume, Language lingua, String arquivo) {
		super(manga, volume, null);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.lingua = lingua;
		this.arquivo = arquivo == null ? "" : arquivo;
		this.capitulos = new ArrayList<MangaCapitulo>();
		this.vocabularios = new HashSet<VocabularioExterno>();
		this.processado = false;
		this.processar = true;
		this.capa = null;
	}

	public MangaVolume(UUID id, String manga, Integer volume, Language lingua, String arquivo, List<MangaCapitulo> capitulos) {
		super(manga, volume, null);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.lingua = lingua;
		this.arquivo = arquivo == null ? "" : arquivo;
		this.capitulos = capitulos;
		this.vocabularios = new HashSet<>();
		this.processado = false;
		this.processar = true;
		this.capa = null;
	}

	public MangaVolume(UUID id, String manga, Integer volume, Language lingua, String arquivo, Set<VocabularioExterno> vocabularios) {
		super(manga, volume, null);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.lingua = lingua;
		this.vocabularios = vocabularios;
		this.arquivo = arquivo == null ? "" : arquivo;
		this.processado = false;
		this.processar = true;
		this.capitulos = new ArrayList<>();
		this.capa = null;
	}

	public MangaVolume(UUID id, String manga, Integer volume, Language lingua, String arquivo, Set<VocabularioExterno> vocabularios, List<MangaCapitulo> capitulos, MangaCapa capa) {
		super(manga, volume, null);
		this.id = id;
		this.manga = manga;
		this.volume = volume;
		this.lingua = lingua;
		this.capitulos = capitulos;
		this.vocabularios = vocabularios;
		this.arquivo = arquivo == null ? "" : arquivo;
		this.processado = false;
		this.processar = true;
		this.capa = capa;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		if (!super.equals(o)) return false;
		MangaVolume that = (MangaVolume) o;
		return Objects.equals(id, that.id) && Objects.equals(manga, that.manga) && Objects.equals(volume, that.volume) && lingua == that.lingua;
	}

	@Override
	public int hashCode() {
		return Objects.hash(super.hashCode(), id, manga, volume, lingua);
	}

	@Override
	public String toString() {
		return "MangaVolume [id=" + id + ", manga=" + manga + ", volume=" + volume + ", lingua=" + lingua + ", vocabularios=" + vocabularios + ", arquivo=" + arquivo + "]";
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/mangaextractor/MangaVolume.kt
package br.com.fenix.processatexto.model.entities.mangaextractor

import br.com.fenix.processatexto.model.entities.Entity
import br.com.fenix.processatexto.model.entities.Manga
import br.com.fenix.processatexto.model.entities.processatexto.VocabularioExterno
import br.com.fenix.processatexto.model.enums.Language
import com.google.gson.annotations.Expose
import java.util.*


data class MangaVolume(
    private var id: UUID? = null,
    @Expose override var manga: String = "",
    @Expose override var volume: Int = 0,
    @Expose var lingua: Language = Language.PORTUGUESE,
    @Expose var capitulos: MutableList<MangaCapitulo> = mutableListOf(),
    @Expose var vocabularios: MutableSet<VocabularioExterno> = mutableSetOf(),
    @Expose var arquivo: String = "",
    var processado: Boolean = false,
    var capa: MangaCapa? = null
) : Manga(), Entity<UUID?, MangaVolume> {

    init {
        super.isProcessar = true
    }

    override fun getId(): UUID? = id

    fun setId(id: UUID?) {
        this.id = id
    }

    override fun create(id: UUID?): MangaVolume {
        TODO("Not yet implemented")
    }

    fun addCapitulos(capitulo: MangaCapitulo) = capitulos.add(capitulo)

    fun addVocabulario(vocabulario: VocabularioExterno) = vocabularios.add(vocabulario)

    constructor(id: UUID?, manga: String, volume: Int, lingua: Language, arquivo: String) : this() {
        setInitial(manga, volume, 0f)
        this.id = id
        this.lingua = lingua
        this.arquivo = arquivo
    }

    constructor(id: UUID?, manga: String, volume: Int, lingua: Language, arquivo: String, capitulos: MutableList<MangaCapitulo>) : this() {
        setInitial(manga, volume, null)
        this.id = id
        this.manga = manga
        this.volume = volume
        this.lingua = lingua
        this.arquivo = arquivo
        this.capitulos = capitulos
    }

    constructor(id: UUID?, manga: String, volume: Int, lingua: Language, arquivo: String, vocabularios: MutableSet<VocabularioExterno>) : this() {
        setInitial(manga, volume, 0f)
        this.id = id
        this.lingua = lingua
        this.vocabularios = vocabularios
        this.arquivo = arquivo
    }

    constructor(
        id: UUID?,
        manga: String,
        volume: Int,
        lingua: Language,
        arquivo: String,
        vocabularios: MutableSet<VocabularioExterno>,
        capitulos: MutableList<MangaCapitulo>,
        capa: MangaCapa?
    ) : this() {
        setInitial(manga, volume, 0f)
        this.id = id
        this.lingua = lingua
        this.capitulos = capitulos
        this.vocabularios = vocabularios
        this.arquivo = arquivo
        this.capa = capa
    }

    override fun toString(): String {
        return "MangaVolume [id=$id, manga=$manga, volume=$volume, lingua=$lingua, vocabularios=$vocabularios, arquivo=$arquivo]"
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), id, manga, volume, lingua)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as MangaVolume

        if (id != other.id) return false
        if (manga != other.manga) return false
        if (volume != other.volume) return false
        if (lingua != other.lingua) return false

        return true
    }
}